### DISCUSSION TITLE
Concerns Regarding the Deprecation of SVN Support and Lack of Separate Directories for Branches and Tags

### DISCUSSION THREAD

**Discussion by nickhoeferpickpro** [community/community#75441] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Dear GitHub Support Team,

I am writing to express my concern and dissatisfaction with the recent decision to sunset Subversion support on GitHub. As evidenced by the brownouts, the shift seems to overlook a particular use case that SVN handles well: maintaining separate directories for each branch and tag without the need to switch or check out.

While I understand the push towards Git and appreciate the innovations with sparse checkouts and the sparse index, these features do not address the need for physical separation of branches in the file system. In many scenarios, such as CI/CD pipelines and complex build environments, having branches checked out in separate directories can significantly simplify processes. Moreover, with the decreasing cost of storage, disk space is no longer a constraint that necessitates the need for a single working directory.

I urge the GitHub team to consider these points and provide alternatives that cater to this specific requirement. Many users rely on the flexibility that SVN integration offers, and its deprecation without a viable Git-based alternative is a step back in functionality for us.

Thank you for your attention to this matter. I hope to see a solution that accommodates the diverse workflows of all GitHub users.

**Comment by schellingb** [comment]
I fully agree with the voiced concerns and I, too, urge the GitHub team to reconsider the sunsetting.

Early October I posted my discontent regarding the planned sunsetting of Subversion support in the repository discussion #69267.
There also seems to be a similar voice of concern in discussion #72718.  Glad to see I'm not the only one!

Personally I have not yet decided on what to do once Subversion support gets shut down. I deeply rely on it for all my open source contributions. I have considered both writing my own SVN2Git bridge or a custom Git client (based on libgit2). In #69267 I raised the question if it were possible for GitHub to release the source code of the Subversion bridge to the public. Such a release could provide a starting point for custom solutions to let users continue with a similar or identical workflow to what is available now.

**Comment by schellingb** [comment]
![image](https://github.com/community/community/assets/14200249/eb2885c4-4195-4f88-981e-8b5e6e970193)
Real upsetting not being able to contribute code today...
